### PR TITLE
[x2cpg] Refactor ExternalCommand

### PR DESCRIPTION
**Pull request by max-leuthaeuser**
Works with Java Process / ProcessBuilder now. No more scala.sys.process.

### CODE REVIEW

**Comment by max-leuthaeuser** [comment]
@mpollmeier I guess I need your help here. Both the tests for csharp and go are failing on MacOS. These are the only frontends using https://github.com/joernio/joern/blob/2cc61c85ff13bf69556eebe05ef000ace3aafe0e/joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/utils/ExternalCommand.scala#L75. The paths there do look wrong and I don't know if this PR is the source or it only revealed this issue.

**Comment by mpollmeier** [comment]
You mean this output, right? I'll take a look later today. 
```
java.lang.RuntimeException: Process exited with code 1. Output:
fail: DotNetAstGen.Program[0] The path /Users/runner/work/joern/joern/"/var/folders/q5/6nls1_g145s2fl1lbvx2zby40000gn/T/x2cpgTestTmpDir14255234666568448138" does not exist!
```

Thanks for the refactoring btw, I had an urge to do something very similar when I touched this code yesterday but didn't get around to actually doing it.

**Comment by max-leuthaeuser** [comment]
Yes, exactly that.

**Comment by mpollmeier** [comment]
the worst bugs are always those that are caused by one or two bytes diffs :) 
https://github.com/joernio/joern/pull/5017/commits/73f74c16d05c4599daf7cc1b4f8bd41e5cac5dcf

might not be the final working version yet, let's see what happens

**Comment by max-leuthaeuser** [comment]
gosrc2cpg may need the same fix, but lets see.

File: joern-cli/frontends/csharpsrc2cpg/src/main/scala/io/joern/csharpsrc2cpg/utils/DotNetAstGenRunner.scala [hunk 1]
```diff
@@ -63,8 +63,8 @@ class DotNetAstGenRunner(config: Config) extends AstGenRunnerBase(config) {
   override def runAstGenNative(in: String, out: File, exclude: String, include: String)(implicit
     metaData: AstGenProgramMetaData
   ): Try[Seq[String]] = {
-    val excludeCommand = if (exclude.isEmpty) "" else s"-e \"$exclude\""
-    ExternalCommand.run(s"$astGenCommand -o ${out.toString()} -i \"$in\" $excludeCommand", ".")
+    val excludeCommand = if (exclude.isEmpty) Seq.empty else Seq("-e", s"\"$exclude\"")
```
Comment (maltek) [inline joern-cli/frontends/csharpsrc2cpg/src/main/scala/io/joern/csharpsrc2cpg/utils/DotNetAstGenRunner.scala:66]: the exclude also needs the quoting fix

Comment (mpollmeier) [reply joern-cli/frontends/csharpsrc2cpg/src/main/scala/io/joern/csharpsrc2cpg/utils/DotNetAstGenRunner.scala]: 👍🏻 yes, thanks

**Comment by max-leuthaeuser** [comment]
Please do not merge yet. Markus may want to add some things.

**Review by ml86 — APPROVED** [review]

**Pull request merged**